lib/schema-controller.js

```diff
@@ -0,0 +1,169 @@
+'use strict'
+
+const { Schemas } = require('./schemas')
+
+const Ajv = require('ajv')
+const fastJsonStringify = require('fast-json-stringify')
+
+let buildPerformanceValidator
+
+function SchemaController () {
+  this.schemaBucket = new Schemas()
+  this.validatorCompiler = undefined
+  this.serializerCompiler = undefined
+  this.parent = undefined
+}
+
+/**
+ * This method is called once on the fastify server inizialization.
+ * You may use it to set some globals argument as caching
+ */
+SchemaController.prototype.setup = function () {
+  // it is needed once at the server built
+  buildPerformanceValidator = ValidatorSelector()
+}
```

Comment:
why isn't this done in the constructor?

Reply:
Because there is a map that needs to be instantiated only once: the `buildPerformanceValidator` track every ajv instances that are being created across all contexts and it avoid to create a new ajv instance if there is already one created with the same options and external schemas

Reply:
See the other comment thread around ids. I think trying to apply context encapsulation to schemas is unnecessarily making things complicated and violates spec.